django/core/management/commands/shell.py

```diff
@@ -124,7 +153,7 @@ def handle(self, **options):
             and not sys.stdin.isatty()
             and select.select([sys.stdin], [], [], 0)[0]
         ):
-            exec(sys.stdin.read(), globals())
+            exec(sys.stdin.read(), {**self.get_namespace()})
```

Comment:
```suggestion
            exec(options["command"], {**globals(), **self.get_namespace()})
```

Reply:
In this case shouldn't we execute from stdin? Because quoting from the docs, there's a section that asserts: 

> You can also pass code in on standard input to execute it.

Reply:
Yes I'm sorry! That was a bad copy paste
 `exec(sys.stdin.read(), {**globals(), **self.get_namespace()})`